withRenew.js

```diff
@@ -45,44 +45,49 @@ const withRenew = WrappedComponent =>
     }
 
     renew(loan, patron, bulkRenewal) {
+      this.setState({ bulkRenewal });
       const params = {
         itemBarcode: loan.item.barcode,
         userBarcode: patron.barcode,
       };
 
-      this.setState({ bulkRenewal });
-
-      return this.props.mutator.renew.POST(params).catch(resp => {
-        const contentType = resp.headers.get('Content-Type');
-        if (contentType && contentType.startsWith('application/json')) {
-          resp.json().then(error => this.handleErrors(error));
-        } else {
-          // eslint-disable-next-line no-alert
-          resp.text().then(error => alert(error));
-        }
-        throw new Error(resp);
+      return new Promise((resolve, reject) => {
+        this.props.mutator.renew.POST(params).then(resolve).catch(resp => {
+          const contentType = resp.headers.get('Content-Type');
+          if (contentType && contentType.startsWith('application/json')) {
+            resp.json().then((error) => {
+              const errors = this.handleErrors(error);
+              reject(this.getMessage(errors));
+            });
+          } else {
+            resp.text().then((error) => {
+              reject(error);
+              alert(error); // eslint-disable-line no-alert
```

Comment:
IIRC, failed renewals will throw a 422 Unprocessable Entity, or may start throwing them as I've seen the backend starting to standardise around that return code. So a failed renewal would throw up an `alert()` which doesn't seem great, especially since you're already showing failed renewals in the UI, right?

Reply:
@doytch You can find partly relevant discussion here https://github.com/folio-org/ui-checkout/pull/131#issuecomment-393168957. So basically, yes, the 422 validation responses are sent as a JSON responses and they are shown in the modal popups, but the other errors such as statusCode >= 500, are sent as plain text which is shown as an alert box. I am not sure if there has been discussion somewhere where it has been decided otherwise.

Reply:
Ahhh weird. Way to sleuth 🔍